lib/persistence/database_provider.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/services.dart';
+import 'package:path/path.dart';
+import 'package:sqflite/sqflite.dart';
+
+class DatabaseProvider {
+  Database? _database;
+  static final DatabaseProvider _instance = DatabaseProvider._();
+  DatabaseProvider._();
+
+  static DatabaseProvider get instance => _instance;
+
+  Future<Database?> getDatabase() async {
+    _database ??= await _connect();
+    return _database!;
+  }
+
+  static Future<Database> _connect() async {
+    print("Database initiated");
+
+    var dbDir = await getDatabasesPath();
+    var dbPath = join(dbDir, "database.sqlite");
+    List<String> commands = await _splitCommands("assets/database/create_db.sql");
+
+    // Delete any existing database:
+    await deleteDatabase(dbPath);
+
+    Database database = await openDatabase(dbPath, version: 1, onCreate: (Database db, int version) async {
+      for (String command in commands) {
+        await db.execute(command);
```

Comment:
zastanawia mnie wydajność tego rozwiązania. Z tego co widziałem, sqflite niestety nie wspiera egzekucji pliku, ale wspiera za to batche. Czy nie lepiej było by w skrypcie dodać commit; po każdej grupie insertów do jednej tabelki i potem traktować tutaj commit; jako wyznacznik końca komendy?

Reply:
Przeniosłem do #25